server/src/api/v1.py

```diff
@@ -157,8 +157,31 @@ def check_token_queue(auth_token: str, secret_key: str, queue: str) -> bool:
     return queue in allowed_queues
 
 
+def check_token_reservation_timeout(
+    auth_token: str, secret_key: str, reservation_timeout: int, queue: str
+) -> bool:
+    """
+    Checks if the requested reservation is either less than the max
+    or that their token gives them the permission to use a higher one
+    """
+    # Max reservation time defaults to 6 hours
+    max_reservation_time = 6 * 60 * 60
+    if reservation_timeout <= max_reservation_time:
+        return True
+    decoded_jwt = decode_jwt_token(auth_token, secret_key)
+    max_reservation_time_dict = decoded_jwt.get("max_reservation_time", {})
+    queue_reservation_time = max_reservation_time_dict.get(queue, 0)
+    star_reservation_time = max_reservation_time_dict.get("*", 0)
+    max_reservation_time = max(queue_reservation_time, star_reservation_time)
```

Comment:
max_reservation_time should be 6 * 60 * 60 if you look a couple lines above. First we check whether or not the requested reservation time is less than that.

Reply:
If the user has requested a reservation time of over 6 hours then we will reach this code and if there are no entries for `*` or this specific queue then `max_reservation_time` will be zero. Which is fine, in a way, because the job will be dropped but I find the logic hard to read.